sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java

```diff
@@ -3,112 +3,89 @@
-                systemProperties.put("azure.keyvault.aad-authentication-url", aadAuthenticationUrl);
-            }
-
-            String clientId = environment.getProperty("azure.keyvault.client-id");
-            if (clientId != null) {
-                systemProperties.put("azure.keyvault.client-id", clientId);
-            }
-
-            String clientSecret = environment.getProperty("azure.keyvault.client-secret");
-            if (clientSecret != null) {
-                systemProperties.put("azure.keyvault.client-secret", clientSecret);
-            }
+        if (environment.getProperty("azure.keyvault.uri") == null) {
+            return;
+        }
 
-            String managedIdentity = environment.getProperty("azure.keyvault.managed-identity");
-            if (managedIdentity != null) {
-                systemProperties.put("azure.keyvault.managed-identity", managedIdentity);
+        putEnvironmentPropertyToSystemProperty(environment, "azure.keyvault.aad-authentication-url");
+        putEnvironmentPropertyToSystemProperty(environment, "azure.keyvault.uri");
+        putEnvironmentPropertyToSystemProperty(environment, "azure.keyvault.tenant-id");
+        putEnvironmentPropertyToSystemProperty(environment, "azure.keyvault.client-id");
+        putEnvironmentPropertyToSystemProperty(environment, "azure.keyvault.client-secret");
+        putEnvironmentPropertyToSystemProperty(environment, "azure.keyvault.managed-identity");
+
+        MutablePropertySources propertySources = environment.getPropertySources();
+        if (KeyVaultKeyStore.KEY_STORE_TYPE.equals(environment.getProperty("server.ssl.key-store-type"))) {
+            Properties properties = new Properties();
+            properties.put("server.ssl.key-store", "classpath:keyvault.dummy");
+            if (hasEmbedTomcat()) {
+                properties.put("server.ssl.key-store-type", "DKS");
             }
-
-            String keyStoreType = environment.getProperty("server.ssl.key-store-type");
-
-            if (keyStoreType != null && keyStoreType.equals("AzureKeyVault")) {
-                MutablePropertySources sources = environment.getPropertySources();
-                Properties properties = new Properties();
-                properties.put("server.ssl.key-store", "classpath:keyvault.dummy");
-
-                try {
-                    Class.forName("org.apache.tomcat.InstanceManager");
-                    properties.put("server.ssl.key-store-type", "DKS");
-                } catch (ClassNotFoundException ex) {
-                }
-
-                PropertiesPropertySource propertySource
-                        = new PropertiesPropertySource("KeyStorePropertySource", properties);
-                sources.addFirst(propertySource);
+            propertySources.addFirst(new PropertiesPropertySource("KeyStorePropertySource", properties));
+        }
+        if (KeyVaultKeyStore.KEY_STORE_TYPE.equals(environment.getProperty("server.ssl.trust-store-type"))) {
```

Comment:
need another variable like TRUST_STORE_TYPE?

Reply:
No. It's just a key store name. No mater trust or not trust, we can use one store name.